powertools-large-messages/src/main/java/software/amazon/lambda/powertools/largemessages/internal/LargeMessageProcessor.java

```diff
@@ -0,0 +1,121 @@
+package software.amazon.lambda.powertools.largemessages.internal;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.exception.SdkException;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.lambda.powertools.largemessages.LargeMessageConfig;
+import software.amazon.lambda.powertools.largemessages.LargeMessageProcessingException;
+import software.amazon.payloadoffloading.S3BackedPayloadStore;
+import software.amazon.payloadoffloading.S3Dao;
+
+import static java.lang.String.format;
+
+/**
+ * Abstract processor for Large Messages. Handle the download from S3 and replace the actual S3 pointer with the content
+ * of the S3 Object leveraging the payloadoffloading library.
+ *
+ * @param <T> any message type that support Large Messages with S3 pointers
+ *            ({@link com.amazonaws.services.lambda.runtime.events.SQSEvent.SQSMessage} and {@link com.amazonaws.services.lambda.runtime.events.SNSEvent.SNSRecord} at the moment)
+ */
+public abstract class LargeMessageProcessor<T> {
+    protected static final String RESERVED_ATTRIBUTE_NAME = "ExtendedPayloadSize";
```

Comment:
Is this key available somewhere from within `payloadoffloading-common` ? Seems like something that all the users of the large messaging pattern need to agree upon

Reply:
Almost, it is in the extended client library (sns and sqs), look at [this](https://github.com/awslabs/amazon-sqs-java-extended-client-lib/blob/master/src/main/java/com/amazon/sqs/javamessaging/SQSExtendedClientConstants.java#L20-L22).